### DISCUSSION TITLE
Couldn't find onAnimationRepeat listener?

### DISCUSSION THREAD

**Discussion by ArtemMikhaylov** [airbnb/lottie-ios#1851] [Q&A] [answered]
Is there some kind of `onAnimationRepeat` listener like in lottie android?  

When animation is in a loop mode, every moment it starts over we have `onAnimationRepeat` called in lottie android. And I couldn't find such possibility in lottie-ios.

It would be nice to have one!

**Comment by calda** [comment, marked as answer]
We don't currently provide callbacks for this type of thing. One key obstacle is that this wouldn't work with the new default Core Animation rendering engine, where Lottie itself isn't running code every frame so wouldn't know when to call the closure.

One way to accomplish this functionality in user code could be:

```swift
func playAnimationOnce() {
  animationView.play(loopMode: .playOnce, completion: {
    // perform code now that the animation has finished its loop
    self.playAnimationOnce() // continue looping after the animation completes
  })
}
